CMakeLists.txt

```diff
@@ -55,6 +55,9 @@ if(NOT FREECAD_LIBPACK_USE OR FREECAD_LIBPACK_CHECKFILE_CLBUNDLER)
         # OPENGL_gl_LIBRARY empty that results into linker errors
         SetupOpenGL()
     endif(BUILD_GUI)
+    if (FREECAD_USE_EXTERNAL_SMESH)
+    	find_package(SMESH REQUIRED)
+    endif(FREECAD_USE_EXTERNAL_SMESH)
```

Comment:
why not move this to SetupSalomeSMESH?

Reply:
One final change, I had excluded `BUILD_SMESH` if `FREECAD_USE_EXTERNAL_SMESH` was enabled, but changed this to allow both in the event unit testing or something else was desired when both were on.

This is identical to the previous logic anyway.

Also tabs->spaces.

Reply:
Inside SetupSalomeSMESH.cmake we have the line `find_package(SMESH CONFIG)` (line 121) that searches for a file SMESHConfig.cmake or smesh-config.cmake.

In my local (external) SMESH build there is a SMESHConfig.cmake file. Don't you have this file or why do you need the `find_package(SMESH REQUIRED)`?

Reply:
When I attempted to build using the existing CMake script when `FREECAD_USE_EXTERNAL_SMESH=ON` was set, it did not seem to appropriately initialize the required SMESH variables. I had to add that extra case.  Perhaps `BUILD_SMESH` was not being set. The variable is a bit ambiguous, as when `FREECAD_USE_EXTERNAL_SMESH` is enabled, SMESH is indeed usually *not* being built during the build process.